### PR TITLE
prov/shm: Handle 0 byte transfer in SAR protocol

### PR DESCRIPTION
**Pull request by shijin-aws**
Currently, SAR protocol doesn't handle 0 byte transfer because it fails to update the resp->status correctly in this corner case. This patch fixes this issue.

Also added test to verify the fix.

The issue can be reproduced by running 

```
fi_rma_bw -p shm -I 5 -v -S 0 -e rdm -s localhost
```

`-v` is needed to make fi_rma_bw run with FI_DELIVERY_COMPLETE, which will force SHM to use SAR protocol.

### CODE REVIEW

File: fabtests/test_configs/shm/shm.exclude [hunk 1]
```diff
@@ -7,7 +7,6 @@ inject_test
 -e dgram
 
 # Exclude tests that use sread/polling
--S
```
Comment (shijin-aws) [inline fabtests/test_configs/shm/shm.exclude:10]: I am not sure if any test is using `-S` for sread. This may need change if that is true

**Review by aingerson — APPROVED** [review]
Thanks!

**Comment by shijin-aws** [comment]
bot:aws:retest

File: prov/shm/src/smr_ep.c [hunk 2]
```diff
@@ -549,6 +549,10 @@ static int smr_format_sar(struct smr_ep *ep, struct smr_cmd *cmd,
 	pending->bytes_done = 0;
 	pending->next = 0;
 
+	/* Nothing to copy for 0 byte transfer */
+	if (!cmd->msg.hdr.size)
+		return 0;
```
Comment (shefty) [inline prov/shm/src/smr_ep.c:554]: Can you update the commit message to indicate *why* SAR needs to handle a 0-byte transfer?  Because on the surface it doesn't make sense.

Comment (shijin-aws) [reply prov/shm/src/smr_ep.c:554]: Yeah, it is actually caught by a corner case when I run `fi_rma_bw -p shm -I 5 -v -S 0 -e rdm -s localhost`. In this test, data validation is enabled, and it forces SHM to run with FI_DELIVERY_COMPLETE (DC), and turn off inject. SHM will use SAR always for DC, so if you run fi_send/recv with SHM + DC + 0 byte, this bug can be triggered. I don't know if any real application is doing that though

Comment (shefty) [reply prov/shm/src/smr_ep.c:554]: I'd like that information in the commit message, so that's it's captured in the git logs, not the PR text.

Comment (shijin-aws) [reply prov/shm/src/smr_ep.c:554]: Done.

Comment (shijin-aws) [reply prov/shm/src/smr_ep.c:554]: @shefty do you have more comments?

**Comment by shijin-aws** [comment]
Is Intel CI failure related?

**Comment by aingerson** [comment]
@shijin-aws We had some systems come up incorrectly after the reboot, you can ignore these failures. shm tests passed

**Review by shefty — APPROVED** [review]
